rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/nodesource/dataspace/DataSpaceNodeConfigurationAgent.java

```diff
@@ -103,4 +285,33 @@ public BooleanWrapper closeNodeConfiguration() {
         return new BooleanWrapper(true);
     }
 
+    /**
+     * This class contains the logic of the cleaning mechanism
+     */
+    private static class CleanTimerTask extends TimerTask {
+
+        @Override
+        public void run() {
+            try {
+                long invalidationPeriod = getCacheInvalidationPeriod();
+                long currentTime = System.currentTimeMillis();
+                // lock the timer in write mode, this will prevent any Task to start during the cleaning process
+                cacheCleaningRWLock.writeLock().lockInterruptibly();
+                try {
+                    FileObject rootFO = fileSystemManager.resolveFile(rootCacheUri);
+                    FileObject[] files = rootFO.findFiles(Selectors.EXCLUDE_SELF);
+                    for (FileObject file : files) {
+                        if (currentTime - file.getContent().getLastModifiedTime() > invalidationPeriod) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS134)